src/deluge/gui/ui/load/load_song_ui.cpp

```diff
@@ -284,7 +284,8 @@ void LoadSongUI::performLoad(StorageManager& bdsm) {
 
 someError:
 		display->displayError(error);
-		storageManager.closeFile();
+		f_close(&smDeserializer.readFIL);
```

Comment:
This feels like it should still be done through the storage manager

Reply:
I moved file closing into the FileReader and FileWriter classes.